oneapi-src/oneAPI-samples
# [MODIN]: Add time results verifier for getting started notebook

# Existing Sample Changes
## Description

[NOTE: This PR is a copy of the accidentally closed #843 (can't be reopened)]

The notebook is supposed to show the performance improvement after switching from stock Pandas to Modin, it compares the execution times of Pandas and Modin in various scenarios. The previous version of the notebook didn't imply that the Modin could be slower than Pandas anyhow and so confused users if such happened.
 
The changes in this PR introduce a time verifier, that compares Modin and Pandas execution times and prints a note if Modin appears to be slower with the possible reasons for it (poor CPU performance, old Modin version, silly environment variables).

Fixes [Jira MODIN6-135](https://jira.devtools.intel.com/browse/MODIN6-135).

## External Dependencies

The changes do not add any new dependencies.

## Type of change

- [x] Implement fixes for Jiras

## How Has This Been Tested?

- [x] Command Line

To test the changes locally:
1. Combine steps from the [`sample.json`](https://github.com/oneapi-src/oneAPI-samples/blob/master/AI-and-Analytics/Getting-Started-Samples/IntelModin_GettingStarted/sample.json) into a single bash script.
<details><summary>script</summary>

```
set -e # Terminate the script on first error
source $(conda info --base)/etc/profile.d/conda.sh # Bypassing conda's disability to activate environments inside a bash script: https://github.com/conda/conda/issues/7980
conda create -y -n intel-aikit-modin intel-aikit-modin -c intel
conda activate intel-aikit-modin
pip install -r requirements.txt # Installing notebook's dependencies
pip install runipy # Installing 'runipy' for extended abilities to execute the notebook
runipy IntelModin_GettingStarted.ipynb # Test 'Modin is faster than Pandas' case
MODIN_CPUS=1 runipy IntelModin_GettingStarted.ipynb # Test 'Modin is slower than Pandas' case
```

</details>

2. Run the script from the sample's directory and ensure that the last notebook's cell printed"[CODE_SAMPLE_COMPLETED_SUCCESFULLY]"

Would also like to get a review from Modin folks: @Garra1980 @YarShev @vnlitvinov
